AIO/src/org/aio/script/AIO.java

```diff
@@ -133,6 +218,22 @@ private void executeTasks() throws InterruptedException {
         }
     }
 
+    /**
+     * Given the start and end indices, pull a new task queue for the current loop
+     *
+     * @return a queue of tasks
+     */
+    private Queue<Task> getLoopTaskQueue(){
+        ArrayList<Task> allOrderedTasks = gui.getTasksAsList();
```

Comment:
Would it be better to just call this once in `loadTasksFromGui`, and store it as a global variable? I don't think we need to do this every time

Reply:
Actually we need a whole new instance of the task each time we execute the loop, that way the task and activity "state" don't leak over into the future executions.

Reply:
Good point.